### ISSUE TITLE
Continuous Integration Meta Issue

### ISSUE THREAD

**Issue by ejgallego** [math-comp/math-comp#159]
The purpose of this issue is to track some planned improvements to math-comp CI testing. In particular, more comprehensive testing of developments is sought as to be able to assess the impact of more aggressive changes.

The general idea here is to reuse Coq's work on that direction, which is planned to use a custom set of scripts and GitLab's pipelines. A rough todo list is: 

- [x] Implement Gitlab mirroring, add basic `.gitlab-ci.yml`
- [x] Select contributions / users to test.
- [x] Wait for Coq's GitLab setup to be up, adapt to math-comp.

**Comment by CohenCyril** [comment]
Hi @ejgallego would you like to follow-up on this?

**Comment by ejgallego** [comment]
We still haven't finished https://github.com/coq/coq/issues/7753 , and in particular we still plan to do some fundamental changes to the setup.

Even so, I guess you folks could benefit from having a similar setup that the one in Coq.

**Comment by Zimmi48** [comment]
@CohenCyril I'm planning to start working again on setting up CI with Nix for coq-community packages in the next couple of weeks. If you are interested, I could also submit something for math-comp.

**Comment by CohenCyril** [comment]
> @CohenCyril I'm planning to start working again on setting up CI with Nix for coq-community packages in the next couple of weeks. If you are interested, I could also submit something for math-comp.

@Zimmi48 I totally forgot about that... what did you do in the end?

**Comment by Zimmi48** [comment]
I just wrote a summary about this here: https://github.com/coq-community/manifesto/issues/37
tl;dr: CI with Nix works fine but is not ready yet for pushing artifacts that can be reused to build depending packages.

**Comment by ejgallego** [comment]
Sorry folks I'm pretty busy this days and not being very helpful here.

A small note, as seen in #265 it seems that sometimes travis is doing a double build of pr / branch. This is due to setting both branches and PRs to be built, and the most convenient fix is to go to the Travis interface and tell it to build only branches that are whitelisted.

**Comment by erikmd** [comment]
Hi @ejgallego!
Actually math-comp CI is about to migrate to GitLab CI very soon (sorry I wanted to open a PR on this today now that the mirror https://gitlab.com/math-comp/math-comp is set up, but I'll need to postpone it for tomorrow wednesday)
Given this move to GitLab CI, I don't know if a Travis CI configuration should be ultimately kept at the same time? maybe that would be a bit redundant.

> it seems that sometimes travis is doing a double build of pr / branch

but that is not redundant: these are two different Travis builds that occur when the PR comes from a local branch − not a fork (the "branch" build takes the very same version of the branch's code, and the "pr" build do the same, but after doing roughly `git checkout master && git merge source-branch`)

**Comment by ejgallego** [comment]
Hi @erikmd , it seems to me that this issue can basically be closed?

**Comment by erikmd** [comment]
Hi @ejgallego, yes indeed! (and feel free to check the 3 boxes in your initial post which are now addressed :)

**Comment by CohenCyril** [comment]
Thank you @erikmd @ejgallego @Zimmi48 (and whoever contributed that I did not name)
